### DISCUSSION TITLE
How to set up next-auth with separate backend running apollo and prisma?

### DISCUSSION THREAD

**Discussion by izaacdb** [nextauthjs/next-auth#2591] [Help]
We have an existing application with a frontend built on nextjs 11.1, and are trying to implement an email-only login system using next-auth. We can send password reset / registration emails via aws.

The application communicates with a server via apollo client. It does not use any nextjs api routes for this.

The server is running apollo, with prisma as an orm, and type-graphql to resolve the queries.

So far I've read the documentation and ended up at https://next-auth.js.org/adapters/prisma
I'm not sure what to do after reading this.

I have a few questions:

- Do we write resolvers for the User, Account, VerificationRequest and Session models? 
- How does next-auth get this data? Do I need to write some apollo queries for the frontend?
- Is there a tutorial for setting up next-auth with prisma, apollo, and email?

**Comment by balazsorban44** [comment]
If I understand correctly, you have two options here. Either create a custom adapter that communicates with your graphql endpoint receiving the User, Account, Session and VerificationRequest data, or you could connect directly with Prisma, and use the official adapter.

https://github.com/nextauthjs/adapters/tree/main/packages/prisma/src/index.ts

Keep in mind we are in a middle of a major release, the adapter API will be drastically simplified, in case you go with implementing your own. 

For comparison:

https://github.com/nextauthjs/adapters/tree/next/packages/prisma/src/index.ts


We don't have full migration docs yet, here is the WIP https://next-auth-git-ndom91-add-migration-docs-nextauthjs.vercel.app/getting-started/upgrade-v4

**Reply by izaacdb** [reply]
@balazsorban44 Thanks for showing me those files I think I understand how the adapters will work with my setup now.

We're probably not releasing our application for a month or two yet - are you expecting to do a full release of v4 soon? If so I'll go with that version

**Reply by balazsorban44** [reply]
Yes, I am hoping to release something in a few weeks at most. Also, feedback on v4 is very welcome, please tell me how it goes!

**Reply by izaacdb** [reply]
Gave it a go - ran into an issue pretty quickly with `"next-auth": "^4.0.0-next.24",` it gives me `Module not found: Can't resolve 'next-auth/react'`. Is that the right version to be using?

**Reply by balazsorban44** [reply]
Make sure to use an exact version, so no `^`. unfortunately there is a broken `4.0.0` version (which I wish I could just remove, but you cannot ever delete published npm packages 😕), and the caret will mess things up.
